text/0000-mfa-rollout.md

```diff
@@ -0,0 +1,129 @@
+
+To reduce most of the risk, users would have MFA enabled on the `UI and API` level. However, due to the use of automated systems and scripts, it would be difficult to enforce. The API key improvements will reduce the risk as much as possible.
+
+# Guide-level explanation
+
+This policy will be planned in phases and it will  be critical to send communications (eg. announcements and emails) outlining the policy (especially for Phase 2 and 3).
+
+## Phase 1: API Key and Gem Publishing Improvements (est. Early 2022 - Spring 2022)
+
+Along with the current features being implemented (setting MFA on certain keys and WebAuthn support), here is the guide level explanation of the further features we will implement for API keys.
+
+### Keys scoped to a gem
+
+When creating a new key, the user will have the option to scope the API key to a specific gem. If selected, there will be a dropdown of the owner’s gems they can select from. On the CLI, if they choose the scope to a gem, they will need to type the name of the gem they want to scope it to.
+
+If the user is removed as an owner of a gem, the API key will be marked as invalid on the API keys page. Editing the key would be prohibited and the user can delete the key.
+
+### Setting an expiry on keys
+
+When creating a key, users will be able to select the validity period upon creation of new keys (1 day, 7 days, 30 days, and 90 days) with the default of 30 days. Users won’t be able to edit the expiration after the API key has been created.
+
+When the API key is expired, the API key will be marked as invalid on the API keys page. Editing the key would be prohibited and the user can delete the key.
+
+Up for discussion, but we propose that all existing keys will be subject to a certain expiry time of around 6 months from the time of the rollout, and all new push keys are required to set an expiry date. These expirations would be staggered to avoid a thundering herd effect.
+
+### Push multiple gems at once
+
+When publishing a gem, authors can push multiple gems at once using `gem push <gems> [options]`. If the user has MFA enabled, an OTP code is to be entered once. Only one host may be specified for all gems. The response would include the results of pushing each gem (success or error). This is for gem owners that own many gems and would need to enter an OTP per gem push which can be cumbersome.
+
+## Phase 2: Recommending MFA on Most Downloaded Gems (est. Spring 2022 - Fall 2022)
```

Comment:
For easier reading by us southern hemisphere folks, can the timeline be specified without seasons? eg. April - October 2022.

Reply:
That's a great point. Thanks @yob! We'll update 👍 .